src/WinstonGelfTransporter.ts

```diff
@@ -0,0 +1,137 @@
+import TransportStream from 'winston-transport';
+import log from 'gelf-pro';
+import os from 'os';
+
+type TransporterOptions = {
+  // Winston transporter options
+  level?: string;
+  silent?: boolean;
+  handleExceptions?: boolean;
+  // Graylog communication options
+  version?: string;
+  host?: string;
+  port?: number;
+  protocol?: string;
+  hostName?: string;
+  additional?: Object;
+};
+
+const MIN_PORT: number = 0;
+const MAX_PORT: number = 65535;
+
+export default class WinstonGelfTransporter extends TransportStream {
+  private options: TransporterOptions;
+
+  private logLevels = Object({
+    emerg: 0,
+    alert: 1,
+    crit: 2,
+    error: 3,
+    warn: 4,
+    notice: 5,
+    info: 6,
+    debug: 7
+  });
+
+  private objectSerializer: (o: object) => string;
+
+  readonly gelfClient: any;
+
+  constructor(options: TransporterOptions) {
+    super(options);
+    this.options = options;
+    const logConfig = Object({ fields: {} });
+
+    // Update protocol if provided
+    if (options.protocol) {
+      logConfig.adapterName = options.protocol;
+    }
+    // Update host and port information if provided
+    if (options.port || options.host) {
+      logConfig.adapterOptions = {};
+      if (
+        options.port &&
+        options.port >= MIN_PORT &&
+        options.port < MAX_PORT
+      ) {
+        logConfig.adapterOptions.port = options.port;
+      }
+      if (options.host) {
+        logConfig.adapterOptions.host = options.host;
+      }
+    }
+    logConfig.fields = options.hostName
+      ? { host: options.hostName }
+      : { host: os.hostname() };
+    if (options.additional) {
+      logConfig.fields = {
+        ...logConfig.fields,
+        ...options.additional
+      };
+    }
+    this.objectSerializer = JSON.stringify;
+
+    log.setConfig(logConfig);
+    this.gelfClient = log;
+  }
+
+  /**
+   * This function will return a network connector
+   * for sending GELF messages.
+   * @returns A TCP Client for graylog.
+   */
+  getNetworkConnector(): any {
```

Comment:
It would also be of type `Logger`, however I can't say that the gelfClient is of type logger because of the reason in my last reply about incorrect typings.